presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisyCountGaussianColumnRandomSeedAggregation.java

```diff
@@ -0,0 +1,154 @@
+        AccumulatorStateFactory<CountScaleRandomSeedState> stateFactory = StateCompiler.generateStateFactory(CountScaleRandomSeedState.class, classLoader);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        List<Type> inputTypes = ImmutableList.of(type, DOUBLE, BIGINT);
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, BIGINT.getTypeSignature(), inputTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())),
+                createInputParameterMetadata(type),
+                INPUT_FUNCTION,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                ImmutableList.of(new AccumulatorStateDescriptor(
+                        CountScaleRandomSeedState.class,
+                        stateSerializer,
+                        stateFactory)),
+                BIGINT);
+
+        Class<? extends Accumulator> accumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
+                Accumulator.class,
+                metadata,
+                classLoader);
+        Class<? extends GroupedAccumulator> groupedAccumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
+                GroupedAccumulator.class,
+                metadata,
+                classLoader);
+        return new BuiltInAggregationFunctionImplementation(NAME, inputTypes, ImmutableList.of(intermediateType), BIGINT,
+                true, false, metadata, accumulatorClass, groupedAccumulatorClass);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type type)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, type),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, DOUBLE),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, BIGINT),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(CountScaleRandomSeedState state, Block valueBlock, Block noiseScaleBlock, Block randomSeedBlock, int index)
+    {
+        double noiseScale = DOUBLE.getDouble(noiseScaleBlock, index);
+        if (noiseScale < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Noise scale must be >= 0");
+        }
+        // Update count and retain scale and random seed
+        state.setCount(state.getCount() + 1);
+        state.setNoiseScale(noiseScale);
+        state.setRandomSeed(BIGINT.getLong(randomSeedBlock, index));
+    }
+
+    public static void combine(CountScaleRandomSeedState state, CountScaleRandomSeedState otherState)
+    {
+        state.setCount(state.getCount() + otherState.getCount());
+        state.setNoiseScale(state.getNoiseScale() > 0 ? state.getNoiseScale() : otherState.getNoiseScale()); // noise scale should be > 0
```

Comment:
same question about the noiseScales being the same as each other.

Reply:
(same as above)